### PR TITLE
Add more schemas

### PR DESCRIPTION
**Pull request by Will-Lo**
Adds addBroker, demoteBroker, removeBroker, topicConfiguration, and userTasks endpoints to schema.

Also adds a build script and instructions to generate the documentation

### CODE REVIEW

**Comment by kidkun** [comment]
@Will-Lo  Thanks for the patch. This patch seems failed `RequestParameterTest`, PTAL.

**Comment by Will-Lo** [comment]
@kidkun  I'll fix it up today, thanks for the heads up!

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java [hunk 1]
```diff
@@ -39,6 +39,9 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
+
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java]: nit:redundant empty line.

File: cruise-control/src/yaml/endpoints/addBroker.yaml [hunk 2]
```diff
@@ -0,0 +1,184 @@
+AddBrokerEndpoint:
+  post:
+    operationId: addBroker
+    summary: Move partitions to the specified brokers, according to the specified goals
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/addBroker.yaml]: Can we ensure all the `summary` and `description` fields end with dot for the new `yaml` files added in this PR.

File: cruise-control/src/yaml/endpoints/addBroker.yaml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    parameters:
+      - name: brokerid
+        in: query
+        description: List of target broker ids
+        schema:
+          type: set
+          items:
+            type: integer
+          # default is a null set if no values are provided
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/addBroker.yaml]: it is not allowed to not set brokerid for this endpoint.

File: cruise-control/src/yaml/endpoints/addBroker.yaml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+      - name: replication_throttle
+        in: query
+        description: Upper bound on the bandwidth in bytes per second used to move replicas.
+        schema:
+          type: integer
+          format: int64
+          minimum: 1
+    responses:
+      '200':
+        description: |-
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/addBroker.yaml]: (Also apply to other yaml files added in this PR)can we remove the new line here(to be consistent with existing yaml)

File: cruise-control/src/yaml/endpoints/addBroker.yaml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+          application/json:
+            schema:
+              $ref: '../responses/progressResult.yaml#/ProgressResult'
+          text/plain:
+             schema:
+               type: string
+      # Response for all errors
+      default:
+        description: |-
+          Unexpected error
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/addBroker.yaml]: (Also apply to other yaml files added in this PR)`Unexpected error` -> `Error response.`(to be consistent with existing endpoints)

File: cruise-control/src/yaml/endpoints/demoteBroker.yaml [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    summary: Demote specified brokers.
+    parameters:
+      - name: brokerid
+        in: query
+        description: List of target broker ids
+        schema:
+          type: set
+          items:
+            type: int32
+            # default is a null set if no values are provided
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/demoteBroker.yaml]: it is not allowed to not set brokerid for this endpoint.

File: cruise-control/src/yaml/endpoints/removeBroker.yaml [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    summary: Remove/decomission specified brokers.
+    parameters:
+      - name: brokerid
+        in: query
+        description: List of target broker ids
+        schema:
+          type: set
+          items:
+            type: integer
+          # default is a null set if no values are provided
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/removeBroker.yaml]: it is not allowed to not set brokerid for this endpoint.

File: cruise-control/src/yaml/endpoints/topicConfiguration.yaml [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        description: Whether dry-run the request or not
+        schema:
+          type: boolean
+          default: true
+      - name: topic
+        in: query
+        description: Regular expression to specify subject topics
+        schema:
+          type: string # topics regex
+          default: null
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/topicConfiguration.yaml]: Can we add an example here?

File: cruise-control/src/yaml/endpoints/userTasks.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
+UserTasksEndpoint:
+  get:
+    summary: Get the recent user tasks from cruise control
+    parameters:
+      - name: client_ids
+        in: query
+        description: Comma separated IP addresses to filter the task results Cruise Control report
+        schema:
+          type: string
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/userTasks.yaml]: Can we add an example for this field?

File: cruise-control/src/yaml/endpoints/userTasks.yaml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+      - name: endpoints
+        in: query
+        description: Comma separated endpoints to filter the task results Cruise Control report
+        schema:
+          type: string
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/userTasks.yaml]: Can we add an example for this field?

File: cruise-control/src/yaml/endpoints/userTasks.yaml [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+      - name: entries
+        in: query
+        description: The number of entries to show in the response
+        schema:
+          type: number
+          format: int64
+          default: MAX_INT
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/userTasks.yaml]: Can we add an `minimum` for this field?

File: cruise-control/src/yaml/endpoints/userTasks.yaml [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+      - name: types
+        in: query
+        description: Comma separated HTTP request types to filter the task results Cruise Control report
+        schema:
+          type: string
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/userTasks.yaml]: Can we add an example for this field?

**Comment by Will-Lo** [comment]
@kidkun I addressed the changes, I also changed `allow_capacity_estimation` to be consistent across all endpoints as well as there was a small discrepancy between some endpoints.

File: cruise-control/src/yaml/endpoints/addBroker.yaml [hunk 13]
```diff
@@ -1,31 +1,31 @@
 AddBrokerEndpoint:
   post:
     operationId: addBroker
-    summary: Move partitions to the specified brokers, according to the specified goals
+    summary: Move partitions to the specified brokers, according to the specified goals.
     parameters:
       - name: brokerid
         in: query
-        description: List of target broker ids
+        description: List of target broker ids,
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/addBroker.yaml]: nit: `,` -> `.`

File: cruise-control/src/yaml/endpoints/topicConfiguration.yaml [hunk 14]
```diff
@@ -4,26 +4,27 @@ TopicConfigurationEndpoint:
...
           type: boolean
           default: true
       - name: topic
         in: query
-        description: Regular expression to specify subject topics
+        description: Regular expression to specify subject topics.
         schema:
           type: string # topics regex
           default: null
+          example: topic_[0-9].*
```
Comment (kidkun) [inline cruise-control/src/yaml/endpoints/topicConfiguration.yaml]: nit: can we use URL encoding for this regex, i.e. `topic_%5B0-9%5D.%2A`

**Pull request merged**